src/printer.js

```diff
@@ -288,6 +288,18 @@ function genericPrint(path, options, print) {
           [doc, constantKeyword, node.name].filter(element => element)
         );
       }
+      if (node.storageLocation) {
+        return join(
+          ' ',
+          [
+            doc,
+            node.visibility,
+            constantKeyword,
+            node.storageLocation,
+            node.name
+          ].filter(element => element)
```

Comment:
I think because when you don't explicitly define memory or storage, it is `undefined` and not null, and therefore was being included as the default (of which is typically `memory`). You can see an example of this in the other PR failing test where it was writing memory even when the original did not explicitly include it.

Reply:
Yep, I think that’s the case ^^

Reply:
Sorry to insist, but as far as I know that code should be logically equivalent to the one I posted.

For the record, I made that change and tests still pass. Either I am missing something here, or we need another test for the situation that you mention.

Reply:
I mean:

```javascript
if (c) {
   return [a, b, c, d].filter(x => x)
}
return [a, b, d].filter(x => x)
```

should be exactly the same as

```javascript
return [a, b, c, d].filter(x => x)
```

That filter will ignore both null and undefined values (as well as all other falsy values, of course).

Reply:
totally down to use your solution as all the tests are still green; less code FTW always 😊 please update this PR @fvictorio